### DISCUSSION TITLE
Airflow trying to access fonts.googleapis.com for CSS rendering

### DISCUSSION THREAD

**Discussion by geoffo-dev** [apache/airflow#24698] [Q&A]
Hello,

We are trying to use Airflow in a private environment and noticed that this needs to access fonts.googleapis.com to render some of the fonts. As this URL is not allowed, it is causing timeout issues... 

Is there anyway to get around this? We are deploying using the latest helm chart and using airflow version 2.3.0.

Many Thanks
Geoff

**Comment by potiuk** [comment]
I suggest to ask your admin toenable them. Googling (well :) ) reveals that there are a number of things you can do including redirecting the fonts.googleapis.com to localhost https://www.webnots.com/fix-slow-page-loading-waiting-fonts-googleapis-com/

**Comment by geoffo-dev** [comment]
thank - so we have done that and the page renders quickly.

Whilst we can do that now, we do plan to operate it in a completely isolated environment and access to web will be prohibited. As you say there is some stuff we can do - but might be a bit of a nightmare. 

Just wondering if there was a quick win or something done to embed the fonts within the 'stock' docker image

**Reply by potiuk** [reply]
I do not know that much of those, but maybe someone else wil chime in. But I think maybe that is a good opportunity by you or someone from your team to pay back for the free software, find out how to do it and make a PR to implement it. 

Airlfow is created by > 2100 people - mostly users like you who implemented something they needed and contributed it back.  Feel free to do the same.

**Reply by geoffo-dev** [reply]
yeah I guess I am not against that - just outside my level of knowledge and wasn't sure if I was being stupid so I would ask here :-).

Having a bit more of a dig it seems to be embedded within the Flask App Builder which Airflow is based on... 

I will see if there is something that can be done, but I think it might be outside my level of understanding.

**Reply by potiuk** [reply]
You can also ask in FAB repo - Daniel is usually very responsive :)

**Reply by geoffo-dev** [reply]
Cool! Thanks  👍
